File: atcoder/abc146/abc146_b.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception{
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(bf.readLine());
        String S = bf.readLine(); 
        
        char[] arr = S.toCharArray();
        String msg = "";
        for(char i : arr){
            if(i+N > 90){
                int dif =(i+N - 90);
                i = (char)(64+dif);
            }
            else
                i+=N;
            
            msg+= i;
        }
        System.out.println(msg); 
        
    }
}

File: atcoder/abc146/abc146_b.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val input = BufferedReader( InputStreamReader( System.`in` ) );
    val N = input.readLine().toInt();
    val S = input.readLine();
    val hashMap = HashMap<Char, Int>();

    val abecedario = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    var i = 0;
    while( i < abecedario.length ) {
        hashMap.put( abecedario[ i ], i )
        i++;
    }
    i = 0;
    var newWord = "";
    while( i < S.length ) {
        newWord += abecedario[ ( hashMap.get( S[ i ] )!! + N )  % abecedario.length  ] ;
        i++;
    }
    print( newWord )
}